process/smartContract/process.go

```diff
@@ -989,32 +892,35 @@ func (sc *scProcessor) processSCOutputAccounts(
 }
 
 func (sc *scProcessor) updateSmartContractCode(
-	account state.UserAccountHandler,
-	outAcc *vmcommon.OutputAccount,
+	scAccount state.UserAccountHandler,
+	outputAccount *vmcommon.OutputAccount,
 	tx data.TransactionHandler,
 ) error {
-	if len(outAcc.Code) == 0 {
+	if len(outputAccount.Code) == 0 {
 		return nil
 	}
 
-	isDeployment := len(account.GetCode()) == 0
-	if isDeployment {
-		account.SetOwnerAddress(tx.GetSndAddr())
-		account.SetCode(outAcc.Code)
+	isDeployment := len(scAccount.GetCode()) == 0
+	isSenderOwner := bytes.Equal(scAccount.GetOwnerAddress(), tx.GetSndAddr())
+	codeMetadata := vmcommon.CodeMetadataFromBytes(scAccount.GetCodeMetadata())
+	isUpgrade := !isDeployment && isSenderOwner && codeMetadata.Upgradeable
 
-		log.Trace("created SC address", "address", hex.EncodeToString(outAcc.Address))
+	if isDeployment {
+		scAccount.SetOwnerAddress(tx.GetSndAddr())
+		scAccount.SetCodeMetadata(outputAccount.CodeMetadata)
+		scAccount.SetCode(outputAccount.Code)
+		log.Trace("updateSmartContractCode(): created", "address", outputAccount.Address)
 		return nil
 	}
 
```

Comment:
move isSenderOwner and isUpgrade here, as only here they are needed.

Reply:
Moved.